expose total item count + paginationUrl helper

drudge opened the pull request:
I have a need to get the total item count for display while paging. I've exposed it as `itemCount`.

Additionally, I created a `paginationUrl` helper that is passed to the views to allow constructing URLs properly when `baseUrl` already contains a query string.

drudge committed:
expose total item count

drudge committed:
Add paginateUrl helper to allow baseUrl to contain a query string

drudge commented:
@tonymilne any interest in merging this? If not, I'll publish a new module to npm.

BigWillie commented:
+1 for this. I was just looking to see if I could get hold of the total count...

snyk-bot committed:
fix: package.json & .snyk to reduce vulnerabilities

The following vulnerabilities are fixed with an upgrade:
- https://snyk.io/vuln/npm:lodash:20180130


The following vulnerabilities are fixed with a Snyk patch:
- https://snyk.io/vuln/npm:uglify-js:20151024

Latest report for drudge/paginate:
https://snyk.io/test/github/drudge/paginate

Some vulnerabilities weren't fixed or ignored, and so will still fail
the Snyk test report.

drudge committed:
Merge pull request #4 from drudge/snyk-fix-81a22a67

[Snyk] Fix for 3 vulnerable dependency paths

snyk-bot committed:
fix: package.json & .snyk to reduce vulnerabilities


The following vulnerabilities are fixed with a Snyk patch:
- https://snyk.io/vuln/npm:uglify-js:20150824

Latest report for drudge/paginate:package.json:
https://snyk.io/test/github/drudge/paginate?targetFile=package.json

drudge committed:
Merge pull request #5 from drudge/snyk-fix-011f103d

[Snyk] Fix for 1 vulnerable dependency path

snyk-bot committed:
fix: package.json & .snyk to reduce vulnerabilities


The following vulnerabilities are fixed with a Snyk patch:
- https://snyk.io/vuln/SNYK-JS-LODASH-567746

snyk-bot committed:
fix: package.json & .snyk to reduce vulnerabilities


The following vulnerabilities are fixed with a Snyk patch:
- https://snyk.io/vuln/SNYK-JS-LODASH-567746

drudge committed:
Merge pull request #8 from drudge/snyk-fix-b5934d6babb7b030d9e34173bf24c4e8

[Snyk] Fix for 1 vulnerabilities
